### PR TITLE
docs: Add `narwhals.dtypes.Date` docstring example

### LINKED ISSUE #1503: [Doc]: Add a docstring example in `narwhals.dtypes` for Date - beginner friendly
### What type of report is this?

Improvement

### Please describe the issue.

Thank you for contributing to Narwhals
May the Undwerwater Unicorn Magic be with You! 🌊🦄

Add a docstring example for Date - dtype, that will be visible here:
https://narwhals-dev.github.io/narwhals/api-reference/dtypes/#narwhals.dtypes.Date

Check https://github.com/narwhals-dev/narwhals/blob/main/CONTRIBUTING.md
for instructions how to set up your environment

Check https://narwhals-dev.github.io/narwhals/api-reference/dtypes/#narwhals.dtypes.List
as a reference of how it should look like

and https://github.com/narwhals-dev/narwhals/blob/main/narwhals/dtypes.py
as a reference on how to create docstrings

If you need any help just let us know!
Happy Contributing💻🎉

### If you have a suggestion on how it should be, add it below.

_No response_

### PR DESCRIPTION
**Pull request by ontowhee**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #1503
- Closes #1503 

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added
- [x] Documented the changes

### CODE REVIEW

File: narwhals/dtypes.py [hunk 1]
```diff
@@ -698,4 +698,23 @@ def __repr__(self) -> str:
...
-    """Data type representing a calendar date."""
+    """Data type representing a calendar date.
+
+    Examples:
+       >>> import pandas as pd
+       >>> import polars as pl
+       >>> import pyarrow as pa
+       >>> import narwhals as nw
+       >>> from datetime import date, timedelta
+       >>> data = [date(2024, 12, 1) + timedelta(days=1) for d in range(1, 4)]
```
Comment (FBruzzesi) [inline narwhals/dtypes.py:709]: I have a tiny nitpick:

```suggestion
       >>> data = [date(2024, 12, 1) + timedelta(days=d) for d in range(4)]
```

**Review by FBruzzesi — COMMENTED** [review]
Thanks @ontowhee! It looks great! I have a tiny comment, after that I think we can merge 🚀✨

File: narwhals/dtypes.py (hunk 1 above)
Comment (ontowhee) [reply narwhals/dtypes.py]: Thank you! I overlooked this!

**Review by FBruzzesi — APPROVED** [review]
HiThereGIF (2)

**Pull request merged**